### ISSUE TITLE
Unstable onChange behaviour on ContributeDetails

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#1796]
Probably introduced in https://github.com/opencollective/opencollective-frontend/pull/1383
See https://github.com/opencollective/opencollective-frontend/pull/1496

**Component**
`ContributeDetails`

**Description**
When changing the amount with the `Other` input, the `onChange` function is called multiple times with different arguments.

```
details >> {totalAmount: 20000, interval: null}
details >> SyntheticEvent {dispatchConfig: {…}, _targetInst: FiberNode, _dispatchListeners: Array(2), _dispatchInstances: Array(2), nativeEvent: InputEvent, …}
details >> {totalAmount: 20000, interval: null}
```

![peek 07-03-2019 16-15](https://user-images.githubusercontent.com/1556356/53966802-3a8c9380-40f4-11e9-9d09-5c4ae2c0430c.gif)

**Comment by flickz** [comment]
I'm looking into it but this has a lot to do with the Styleguide examples because I couldn't reproduce from the donate / contribute page.

**Comment by Betree** [comment]
@flickz have you tried to debug directly from the style guide with `npm run styleguide:dev` ?

**Comment by flickz** [comment]
Yeah I tried, thinking it has more to do with the `onChange` handler in the styleguide example:
```jsx
<ContributeDetails onChange={console.log} amountOptions={amountOptions} currency={currency} />
```

**Comment by Betree** [comment]
@flickz Styleguide has no difference with the real application, it's putting components in isolation but there's no fundamental difference between rendering in styleguide and rendering in the app.

The handler you show here is not a problem neither, it's a simple `console.log` that will display anything that gets passed to it.

From what I can see, we're passing the `onChange` handler to a component that shouldn't receive it.
Check this line: https://github.com/opencollective/opencollective-frontend/blob/e0fff9d49d9d23ae0c9060bf2a9225e3c3268f81/src/components/ContributeDetails.js#L91

**Comment by flickz** [comment]
@Betree Good catch on this, Thank you.

**Comment by flickz** [comment]
Closed this in [#1516](https://github.com/opencollective/opencollective-frontend/pull/1516)
